auth-jwt/Sources/App/Middleware/BearerAuthenticator.swift

```diff
@@ -0,0 +1,45 @@
+import Foundation
+import Hummingbird
+import HummingbirdAuth
+import JWTKit
+
+struct TokenPayload: HBAuthenticatable {
+  var subject:String
+}
+
+protocol DataProtocol {}
+
+struct BearerAuthenticator: HBAsyncAuthenticator, DataProtocol {
+  var jwksUrl: String
+
+  func authenticate(request: HBRequest) async throws -> TokenPayload? {
+    guard let jwtToken = request.authBearer?.token else { throw HBHTTPError(.unauthorized) }
```

Comment:
In theory authenticators should return `nil` when they fail to authenticate. I guess hummingbird needs an equivalent of the Vapor `GuardMiddleware` to return 404s when a request is not authenticated. I'll add one.